### DISCUSSION TITLE
Parsing `readContracts` with type safety

### DISCUSSION THREAD

**Discussion by kyscott18** [wevm/wagmi#1562] [Ideas]
I am looking to create a wrapper around `readContracts` and `readContract` that accepts an extra parse parameter to take the data returned from the contract and create a more sophisticated type such as type `Token` or `UniswapV2Pair`. 

What is the best way to go about this given the pretty sophisticated types used in `readContracts`?  `readContract` was somewhat straightforward and great to learn more about typescript types https://github.com/kyscott18/fReed/blob/master/src/readContract.ts  but struggling with `readContracts`

Ideally I could add a new property to `contracts` in `ReadContractsConfig` that takes the type returned by the contract read and returns a new type `TResponse`. The return type of the function would be an array of `TResponse`. Any hints on how to get this to work?

**Comment by milaabl** [comment]
Yes, that'd be great if it could be like `useContractRead<IExpectedReturnType>()`.
